osu.Game/Screens/Select/Carousel/CarouselBeatmapRank.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+using osu.Game.Online.Leaderboards;
+using osu.Game.Rulesets;
+using osu.Game.Scoring;
+
+namespace osu.Game.Screens.Select.Carousel
+{
+    public class CarouselBeatmapRank : Container
+    {
+        private const int rank_size = 20;
+        private readonly BeatmapInfo beatmap;
+
+        private TopLocalRank rank;
+
+        public CarouselBeatmapRank(BeatmapInfo beatmap)
+        {
+            this.beatmap = beatmap;
+
+            Height = rank_size;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ScoreManager scores, IBindable<RulesetInfo> ruleset)
+        {
+            scores.ItemAdded += scoreChanged;
+            scores.ItemRemoved += scoreChanged;
+            ruleset.ValueChanged += _ => rulesetChanged();
+
+            rank = new TopLocalRank(beatmap)
+            {
+                ScoreLoaded = scaleDisplay
+            };
+
+            InternalChild = new DelayedLoadWrapper(rank)
+            {
+                RelativeSizeAxes = Axes.Both
+            };
+        }
+
+        private void rulesetChanged()
+        {
+            rank.FetchAndLoadTopScore();
+        }
+
+        private void scoreChanged(ScoreInfo score)
+        {
+            if (score.BeatmapInfoID == beatmap.ID)
+            {
+                rank.FetchAndLoadTopScore();
+            }
+        }
+
+        private void scaleDisplay(ScoreInfo score)
+        {
+            if (score != null)
+                Width = rank_size * 2;
```

Comment:
Not sure what the point of this is really... Loads will not take a considerable amount of time, and this only increases complexity of implementation.

Reply:
Since DrawableRank has RelativeSizeAxes = Axes.Both, the parents need to be sized accordingly. The rank is displayed before the starcounter, and if there is no top rank for the beatmap then it would create empty space between the starcounter and difficultyicon. This was just to scale the width, so that if there is no rank then the width would be 0 also. But I removed this logic in the new commit and changed it to Hide() / Show().